k-distribution/src/main/scripts/bin/krun

```diff
@@ -45,6 +45,21 @@ tempFiles+=("$input_file")
 tempFiles+=("$expanded_input_file")
 tempFiles+=("$kore_output")
 
+keepTempsIfDryRun () {
+  # kepp files used by dry-run
+  if $dryRun; then
+    local newArray=()
+    for value in "${tempFiles[@]}"; do
```

Comment:
Will this work on MacOS? I thought that it only worked in Bash 4+ to have an empty array and then append to it later.

Reply:
I think the problem was with associative arrays. Anyways, I've tested it with the mac version and it works:
```
radu@radu:~/work/t e s t$ cat test.sh
krun --dry-run 1.test
radu@radu:~/work/t e s t$ ../bash/bash-3.2.57/bash test.sh
kore-exec ./test-kompiled/definition.kore --module TEST --pattern .krun-2021-04-15-21-51-20-Z2RhjNb9Yx/tmp.in.C1RvjO4LLV --output .krun-2021-04-15-21-51-20-Z2RhjNb9Yx/result.kore
radu@radu:~/work/t e s t$ ls -a
.  ..  1.test  .krun-2021-04-15-21-51-20-Z2RhjNb9Yx  test.k  test-kompiled  test.sh
```